BedrockServer.cpp

```diff
@@ -772,17 +772,23 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
     // our state right before we commit.
     SQLiteNodeState state = _replicationState.load();
 
-    // If we're following, we will automatically escalate any command that's not already complete (complete
-    // commands are likely already returned from leader with legacy escalation) and is marked as
-    // `escalateImmediately` (which lets them skip the queue, which is particularly useful if they're waiting
-    // for a previous commit to be delivered to this follower), OR if we're on a different version from leader.
-    if (state == SQLiteNodeState::FOLLOWING && (_version != _leaderVersion.load() || command->escalateImmediately) && !command->complete) {
+    // If we're following, we will automatically escalate any command that's:
+    // 1. Not already complete (complete commands are likely already returned from leader with legacy escalation) 
+    // and is marked as `escalateImmediately` (which lets them skip the queue, which is particularly useful if they're waiting
+    // for a previous commit to be delivered to this follower);
+    // 2. Any commands if the current version of the code is not the same one as leader is executing.
+    if (state == SQLiteNodeState::FOLLOWING && !command->complete && (command->escalateImmediately || _version != _leaderVersion.load())) {
         auto _clusterMessengerCopy = _clusterMessenger;
-        if (_clusterMessengerCopy && _clusterMessengerCopy->runOnLeader(*command)) {
+        string escalatedTo = "";
+        if (command->escalateImmediately && _clusterMessengerCopy && _clusterMessengerCopy->runOnPeer(*command, true)) {
             // command->complete is now true for this command. It will get handled a few lines below.
             SINFO("Immediately escalated " << command->request.methodLine << " to leader.");
+            escalatedTo = "leader";
+        } else if (_version != _leaderVersion.load() && _clusterMessengerCopy && _clusterMessengerCopy->runOnPeer(*command, false)) {
+            SINFO("Escalated " << command->request.methodLine << " to follower peer.");
+            escalatedTo = "follower peer";
         } else {
-            SINFO("Couldn't immediately escalate command " << command->request.methodLine << " to leader, queuing normally.");
+            SINFO("Couldn't escalate command " << command->request.methodLine << " to " << escalatedTo << ", queuing it again.");
```

Comment:
Isn't `escalatedTo` always an empty string here?

Reply:
You're right, fixed!